### PR TITLE
Fix & Localize sorting order

### LINKED ISSUE #2: Sorting is broken for full stacks
When an item only has full stacks the last stack gets moved to the end of the inventory. If multiple items meet this condition, each one has a stack moved to the end of the inventory.

There's already a PR open to fix this (#1 ), I'm just adding the issue to keep track until then.

### LINKED ISSUE #3: Sorting is not localized
Sorting is not done by the visible name of the items. For example, Beehive, Bees and Honey are not sorted in that order when using English locale.

This is fixed by PR #1 and I'm just creating the issue to keep track of it until the PR is merged.

### PR DESCRIPTION
**Pull request by dcsobral**

### CODE REVIEW

**Review by made-by-traxam — APPROVED** [review]
Looks good!

**Comment by made-by-traxam** [comment]
I'll test and publish at some point this week

**Comment by dcsobral** [comment]
Though I think this is better than the current version I have identified one area for improvement. While the localized comparison produces better results overall, the plants you can acquire at Tangaroa all have the same localized name in English. I need to make the internal name a fallback when ordering.

**Comment by dcsobral** [comment]
Alright, I fixed the issue with the Tangora Paintings and Tangaroa Plants. I also added ordering by stack size for same items, which should help compaction -- I've observed compaction taking multiple sorts to fully compact before.

**Review by made-by-traxam — APPROVED** [review]

**Comment by made-by-traxam** [comment]
I tested your changes and it seems like the sorting order does not change when you change your game language without restarting the game. I also spent some time finding out why that is:

When the language is changed in the game, for every `Item_Base` registered in the `ItemManager`, the `DisplayName` property of the `settings_Inventory` (type `ItemInstance_Inventory`) attribute is updated to the localized string. This is also the property that is shown within menus. However, Raft's code creates a **copy of** (and not a **reference to**) those settings when instantiating an item instance that is used in the inventory.

That means that the `ItemInstance.settings_Inventory.DisplayName` property is not properly updated when the localization is changed.

You can solve this problem by using `ItemInstance.baseItem.settings_Inventory.DisplayName` instead of `ItemInstance.settings_Inventory.DisplayName` for sorting.

File: SortInventory/SortInventoryMod.cs [hunk 1]
```diff
@@ -122,4 +123,22 @@ public void OnModUnload()
...
         }
     }
-}
\ No newline at end of file
+
+    sealed class ItemComparer : IComparer<ItemInstance>
+    {
+        public int Compare(ItemInstance x, ItemInstance y)
+        {
+            var localized = String.Compare(x.settings_Inventory.DisplayName, y.settings_Inventory.DisplayName, StringComparison.CurrentCultureIgnoreCase);
```
Comment (made-by-traxam) [inline SortInventory/SortInventoryMod.cs]: `ItemInstance.settings_Inventory.DisplayName` is not properly changed when updating the language. Use `ItemInstance.baseItem.settings_Inventory.DisplayName` instead.

**Review by made-by-traxam — CHANGES_REQUESTED** [review]

File: SortInventory/SortInventoryMod.cs (hunk 1 above)
Comment (dcsobral) [reply SortInventory/SortInventoryMod.cs]: Got it. I'll update this.

**Review by made-by-traxam — APPROVED** [review]

**Comment by made-by-traxam** [comment]
Merged and released! I'll add your name to the contributors list, thanks!

**Pull request merged**